src/StrawberryShake/Client/src/Core/Serialization/UuidSerializer.cs

```diff
@@ -5,11 +5,25 @@ namespace StrawberryShake.Serialization
     /// <summary>
     /// This serializer handles UUID scalars.
     /// </summary>
-    public class UUIDSerializer : ScalarSerializer<Guid>
+    public class UUIDSerializer : ScalarSerializer<string, Guid>
     {
         public UUIDSerializer(string typeName = BuiltInScalarNames.UUID)
             : base(typeName)
         {
         }
+
+        public override Guid Parse(string serializedValue)
+        {
+            if (Guid.TryParse(serializedValue, out Guid guid))
+            {
+                return guid;
+            }
+            throw ThrowHelper.UuidSerializer_CouldNotParse(serializedValue);
+        }
+
+        protected override string Format(Guid runtimeValue)
+        {
+            return runtimeValue.ToString();
```

Comment:
We should use a specific format here. We could have a optional format in the constructor.

Reply:
I updated the UuidSerializer to support optionally other formats.